cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -1067,13 +1143,15 @@ val Node?.assigns: List<AssignExpression>
  *
  * @param predicate the search predicate
  */
-fun Node.firstParentOrNull(predicate: (Node) -> Boolean): Node? {
+inline fun <reified T : Node> Node.firstParentOrNull(
```

Comment:
This change is so awesome! Can we extract this please into a separate PR because I think we are using this pattern in many more places

Reply:
Of course. When I do so, I'd also address my comment above on whether we should immediately start in `node.astParent` or if we should return `node` immediately if it fulfills the `predicate` and is of `T`

Reply:
or should I introduce 2 functions?

Reply:
See #2018